Improve build Docker image on Alpine Linux with musl libc

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-kvrocks/issues) and found no similar issues.


### Motivation

If I need a build kvrocks Docker image, based on Alpine linux, I had an error:


> #0 3841.6 [ 98%] Building CXX object CMakeFiles/kvrocks.dir/src/main.cc.o
> #0 3851.1 [100%] Linking CXX executable kvrocks
> #0 3994.4 /usr/lib/gcc/x86_64-alpine-linux-musl/11.2.1/../../../../x86_64-alpine-linux-musl/bin/ld: CMakeFiles/kvrocks.dir/src/main.cc.o: in function `SegvHandler':
> #0 3995.4 /kvrocks/src/main.cc:87: undefined reference to `backtrace'
> #0 3995.4 /usr/lib/gcc/x86_64-alpine-linux-musl/11.2.1/../../../../x86_64-alpine-linux-musl/bin/ld: /kvrocks/src/main.cc:88: undefined reference to `backtrace_symbols'
> #0 4004.7 collect2: error: ld returned 1 exit status
> #0 4004.7 make[3]: *** [CMakeFiles/kvrocks.dir/build.make:249: kvrocks] Error 1
> #0 4004.7 make[2]: *** [CMakeFiles/Makefile2:521: CMakeFiles/kvrocks.dir/all] Error 2
> #0 4004.8 make[1]: *** [CMakeFiles/Makefile2:528: CMakeFiles/kvrocks.dir/rule] Error 2
> #0 4004.8 make: *** [Makefile:254: kvrocks] Error 2

It seems like musl does not provide the backtrace function.

### Solution

Some about this I fined there - https://github.com/xournalpp/xournalpp/issues/3817 and now I test some fix. 


### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -37,6 +37,8 @@ if (CMAKE_VERSION VERSION_GREATER_EQUAL "3.24.0")
     cmake_policy(SET CMP0135 NEW)
 endif()
 
+find_package(Backtrace REQUIRED)
+
 if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
     if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7)
         message(FATAL_ERROR "It is expected to build kvrocks with GCC 7 or above")
@@ -151,6 +153,7 @@ endif()
 list(APPEND EXTERNAL_LIBS tbb)
 list(APPEND EXTERNAL_LIBS jsoncons)
 list(APPEND EXTERNAL_LIBS Threads::Threads)
+list(APPEND EXTERNAL_LIBS ${Backtrace_LIBRARY})
 
 # Add git sha to version.h
 find_package(Git REQUIRED)
@@ -185,7 +188,7 @@ list(FILTER KVROCKS_SRCS EXCLUDE REGEX src/main.cc)
 
 add_library(kvrocks_objs OBJECT ${KVROCKS_SRCS})
 
-target_include_directories(kvrocks_objs PUBLIC src src/common ${PROJECT_BINARY_DIR})
+target_include_directories(kvrocks_objs PUBLIC src src/common ${PROJECT_BINARY_DIR} ${Backtrace_INCLUDE_DIR})
 target_compile_features(kvrocks_objs PUBLIC cxx_std_17)
 target_compile_options(kvrocks_objs PUBLIC -Wall -Wpedantic -Wsign-compare -Wreturn-type -fno-omit-frame-pointer -Werror=unused-result)
 if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")

